UTL-387 Upgrade.

halfninja opened the pull request:
The main issue was the import of the Node 'url' module. Webpack 5 stops importing polyfills for these things and they don't exist in the browser.

We only used it for IE11 support, so I've removed all that and the import, as well as some minor tidying up.

Added the 'files' property to package.json, so that we don't include extra junk in the published package.

halfninja committed:
UTL-387 Upgrade.

The main issue was the import of the Node 'url' module. Webpack 5 stops importing polyfills for these things and they don't exist in the browser.

We only used it for IE11 support, so I've removed all that and the import, as well as some minor tidying up.

Added the 'files' property to package.json, so that we don't include extra junk in the published package.

halfninja commented on package.json:
```diff
@@ -1,26 +1,31 @@
-    "@babel/register": "7.0.0",
-    "chai": "4.2.0",
-    "mocha": "5.2.0"
+    "@babel/core": "^7.28.0",
```
We actually don't need Babel any more. brb

halfninja committed:
UTL-387 Remove Babel.

halfninja commented on lib/serverpipe.js:
```diff
@@ -1,33 +1,11 @@
+import { getCsrfHeaderName, getCsrfToken } from './csrfToken.js';
 
 export function fetchWithCredentials(url, options = {}) {
     const headers = 'headers' in options ? options.headers : {};
```
By jove, it might be right you know

halfninja replied:
It wouldn't fail with a null URL but the origin would always match so it might mean addCsrfHeaderIfSafe always adds the headers, even if the requested URL was a different origin.

flamingelk commented on lib/serverpipe.js:
```diff
@@ -1,33 +1,11 @@
-    return ie;
-}
-
-export function addQsToUrl(url, qs) {
```
If and when you release this, can you make it an upgrade-breaking release, as various apps use this function.

halfninja replied:
I've reinstated the function in a dependency-free form

halfninja committed:
URL-387 Reinstate our beloved addQsToUrl.

It is consumed by apps, so let's keep it (without using the url module, or the URL class which only supports full URLs that have an origin)
